### PR TITLE
Server request content streams

### PR DESCRIPTION
**Pull request by DJLemkes**
Handler that adds streaming requests support as proposed in #137.

This handler only creates streams when strictly necessary. Besides that, it allows the user to not consume the entire request before sending out a response.

In terms of performance it is roughly equal to that of non-streaming for small <=6KB requests. For larger requests it tends to be faster than non-streaming for larger request bodies. These observations are only based on my machine and ran with the `HelloWorldAdvanced` example app.

Lastly, it also fixes a bug currently in the `main` branch where subsequent responses were causing Netty to throw exceptions because of `LAST_EMPTY_CONTENT` not being written correctly.

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala [hunk 1]
```diff
@@ -67,8 +67,10 @@ final case class ServerRequestHandler[R](
               } yield ()
             }
           case HttpData.CompleteData(data) =>
-            ctx.writeAndFlush(JUnpooled.copiedBuffer(data.toArray), ctx.channel().voidPromise())
-          case HttpData.Empty              => ctx.flush()
+            ctx.write(JUnpooled.copiedBuffer(data.toArray), ctx.channel().voidPromise())
+            ctx.writeAndFlush(JLastHttpContent.EMPTY_LAST_CONTENT)
+          case HttpData.Empty              =>
+            ctx.writeAndFlush(JLastHttpContent.EMPTY_LAST_CONTENT)
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala:73]: Please move this to separate PR

**Comment by amitksingh1490** [comment]
> Lastly, it also fixes a bug currently in the `main` branch where subsequent responses were causing Netty to throw exceptions because of `LAST_EMPTY_CONTENT` not being written correctly.
I am moving this to separate PR as this is a blocker to run benchmarks @DJLemkes

**Comment by DJLemkes** [comment]
Let me know what you think. More then happy to make changes if required :).

**Comment by tusharmath** [comment]
This is great @DJLemkes! Will have look at it.

**Comment by DJLemkes** [comment]
@tusharmath sorry to tag you again :). I can do another rebase when you're ready to review. And again, happy to make changes. Would it help if I added a description of my benchmark setup (used `go-wrk` to POST request bodies as well).

**Comment by tusharmath** [comment]
@DJLemkes Update: I am still doing some POCs on how to incorporate streaming capabilities to ZIO Http without compromising on performance. Thought I should post it here to keep you also updated :)

**Comment by tusharmath** [comment]
Will get this out soon. We have a design that is promising and reasonably performant.

**Comment by tusharmath** [comment]
Will get this out soon. We have a design that is promising and reasonably performant.
